engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/execution/RuntimeExecutionHooks.java

```diff
@@ -21,15 +22,26 @@ public void add(Runnable hook) {
 
   @Override
   public void run() {
+    Set<Runnable> hooksToRun;
     synchronized (hooks) {
-      for (Runnable hook : hooks) {
+      hooksToRun = new LinkedHashSet<>(hooks);
```

Comment:
I rather use `hooks.toArray(new Runnable[0])` and then immediately `hooks.clear()`.